pkg/eventfilter/subscriptionsapi/cesql_correlationid_filter.go

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subscriptionsapi
+
+import (
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/des"
+	"crypto/rc4"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
+	"regexp"
+	"strings"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cesql "github.com/cloudevents/sdk-go/sql/v2"
+	cefn "github.com/cloudevents/sdk-go/sql/v2/function"
+	ceruntime "github.com/cloudevents/sdk-go/sql/v2/runtime"
+
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+)
+
+// Add a user defined function to validate correlation id, then return a cesql_filter
+func NewCESQLCorrelationIdFilter(ctx context.Context) error {
```

Comment:
nit: this function doesn't actually create the filter anymore, it just registers the function with the runtime
```suggestion
func RegisterCESQLCorrelationIdFilter(ctx context.Context) error {
```